test/suite/test_prepare03.py

```diff
@@ -0,0 +1,211 @@
+# software under copyright law.
+#
+# THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND,
+# EXPRESS OR IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF
+# MERCHANTABILITY, FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT.
+# IN NO EVENT SHALL THE AUTHORS BE LIABLE FOR ANY CLAIM, DAMAGES OR
+# OTHER LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE,
+# ARISING FROM, OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR
+# OTHER DEALINGS IN THE SOFTWARE.
+
+import wiredtiger, wttest
+from wtscenario import make_scenarios
+
+# test_prepre03.py
+#    Prepare transaction check post conditions for cursor operations
+
+# Pattern of test script is to invoke cursor operations in prepared
+# transactions to ensure they fail and to commit transaction and repeat
+# same operations to ensure normally they pass.
+class test_prepre03(wttest.WiredTigerTestCase):
+    """
+    Test basic operations
+    """
+    table_name1 = 'test_prepare_cursor'
+    nentries = 10
+
+    scenarios = make_scenarios([
+        ('file-col', dict(tablekind='col',uri='file')),
+        ('file-fix', dict(tablekind='fix',uri='file')),
+        ('file-row', dict(tablekind='row',uri='file')),
+        ('lsm-row', dict(tablekind='row',uri='lsm')),
+        ('table-col', dict(tablekind='col',uri='table')),
+        ('table-fix', dict(tablekind='fix',uri='table')),
+        ('table-row', dict(tablekind='row',uri='table'))
+    ])
+
+    def genkey(self, i):
+        if self.tablekind == 'row':
+            return 'key' + str(i)
+        else:
+            return long(i+1)
+
+    def genvalue(self, i):
+        if self.tablekind == 'fix':
+            return int(i & 0xff)
+        else:
+            return 'value' + str(i)
+
+    def assertCursorHasNoKeyValue(self, cursor):
+        keymsg = '/requires key be set/'
+        valuemsg = '/requires value be set/'
+        self.assertRaisesWithMessage(
+            wiredtiger.WiredTigerError, cursor.get_key, keymsg)
+        self.assertRaisesWithMessage(
+            wiredtiger.WiredTigerError, cursor.get_value, valuemsg)
+
+    def session_create(self, name, args):
+        """
+        session.create, but report errors more completely
+        """
+        try:
+            self.session.create(name, args)
+        except:
+            print('**** ERROR in session.create("' + name + '","' + args + '") ***** ')
+            raise
+
+    # Create and session and test cursor operations.
+    def test_prepare_cursor(self):
+        tablearg = self.uri + ':' + self.table_name1
+        preparemsg = "/ not permitted in a prepared transaction/"
+        if self.tablekind == 'row':
+            keyformat = 'key_format=S'
+        else:
+            keyformat = 'key_format=r'  # record format
+        if self.tablekind == 'fix':
+            valformat = 'value_format=8t'
+        else:
+            valformat = 'value_format=S'
+        create_args = keyformat + ',' + valformat
```

Comment:
This sort of thing should be part of the scenarios so that you can just access them in one location. You can see an example in `test_timestamp04.py` in the `types` section rather than hardcoded here.